components/match2/wikis/leagueoflegends/matchgroup_input_custom.lua

```diff
@@ -0,0 +1,627 @@
+		matchsection = Variables.varDefault('matchsection'),
+		comment = match.comment,
+		mvp = matchFunctions.getMVP(match),
+	}
+	return match
+end
+
+function matchFunctions.getMVP(match)
+	if not match.mvp then return nil end
+	local mvppoints = match.mvppoints or 1
+
+	-- Split the input
+	local players = mw.text.split(match.mvp, ',')
+
+	-- Trim the input
+	for index, player in pairs(players) do
+		players[index] = mw.text.trim(player)
+	end
+
+	return {players=players, points=mvppoints}
+end
+
+function matchFunctions.getOpponents(match)
+	-- read opponents and ignore empty ones
+	local opponents = {}
+	local isScoreSet = false
+	for opponentIndex = 1, _MAX_NUM_OPPONENTS do
+		-- read opponent
+		local opponent = match['opponent' .. opponentIndex]
+		if not Logic.isEmpty(opponent) then
+			CustomMatchGroupInput.processOpponent(opponent, match.date)
+
+			-- Retrieve icon for team
+			if opponent.type == Opponent.team then
+				opponent.icon, opponent.icondark = opponentFunctions.getIcon(opponent.template)
+			end
+
+			-- apply status
+			opponent.score = string.upper(opponent.score or '')
+			if Logic.isNumeric(opponent.score) then
+				opponent.score = tonumber(opponent.score)
+				opponent.status = _STATUS_SCORE
+				isScoreSet = true
+			elseif Table.includes(_ALLOWED_STATUSES, opponent.score) then
+				opponent.status = opponent.score
+				opponent.score = _NOT_PLAYED_SCORE
+			end
+
+			-- get players from vars for teams
+			if opponent.type == Opponent.team then
+				if not Logic.isEmpty(opponent.name) then
+					match = matchFunctions.getPlayersOfTeam(match, opponentIndex, opponent.name)
+				end
+			elseif opponent.type == Opponent.solo then
+				opponent.match2players = Json.parseIfString(opponent.match2players) or {}
+				opponent.match2players[1].name = opponent.name
+			elseif opponent.type ~= Opponent.literal then
+				error('Unsupported Opponent Type "' .. (opponent.type or '') .. '"')
+			end
+
+			opponents[opponentIndex] = opponent
+		end
+	end
+
+	--apply walkover input
+	match.walkover = string.upper(match.walkover or '')
+	if Logic.isNumeric(match.walkover) then
+		local winnerIndex = tonumber(match.walkover)
+		opponents = matchFunctions._makeAllOpponentsLoseByWalkover(opponents, _STATUS_DEFAULT_LOSS)
+		opponents[winnerIndex].status = _STATUS_DEFAULT_WIN
+		match.finished = true
+	elseif Logic.isNumeric(match.winner) and Table.includes(_ALLOWED_STATUSES, match.walkover) then
+		local winnerIndex = tonumber(match.winner)
+		opponents = matchFunctions._makeAllOpponentsLoseByWalkover(opponents, match.walkover)
+		opponents[winnerIndex].status = _STATUS_DEFAULT_WIN
+		match.finished = true
+	end
+
+	-- see if match should actually be finished if bestof limit was reached
+	if isScoreSet and not Logic.readBool(match.finished) then
+		local firstTo = math.floor(match.bestof / 2)
+		for _, item in pairs(opponents) do
+			if tonumber(item.score or 0) > firstTo then
+				match.finished = true
+				break
+			end
+		end
```

Comment:
are draws possible?
if so you might want to account for them here
maybe something like
```suggestion
		local firstTo = math.floor(match.bestof / 2)
		local isDraw = true
		for _, item in pairs(opponents) do
			local score = tonumber(item.score) or 0
			if score > firstTo then
				match.finished = true
				break
			elseif score < firstTo then
				isDraw = false
			end
		end
		if isDraw then
			match.finished = true
		end
```

Reply:
No map draws to my knowledge

Reply:
this is for the match not the map

with the current code i do not see it auto finishing for match draws

Reply:
Code updated, though I did it slightly differently.

Reply:
yeah that works too :)